src/framework/preferences/js/ArrowScrolling.js

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2017 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var fluid_3_0_0 = fluid_3_0_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    /************************************************************************************
+     * Scrolling Panel Prefs Editor:                                                    *
+     * This is a mixin grade to be applied to a fluid.prefs.prefsEditor type component. *
+     * Typically used for responsive small screen presentations of the separated panel  *
+     * to allow for scrolling by clicking on left/right arrows                          *
+     ************************************************************************************/
+
+    fluid.defaults("fluid.prefs.arrowScrolling", {
+        selectors: {
+            scrollContainer: ".flc-prefsEditor-scrollContainer"
+        },
+        listeners: {
+            "onReady.windowResize": {
+                "this": window,
+                method: "addEventListener",
+                args: ["resize", "{that}.events.onSignificantDOMChange.fire"]
+            }
+        },
+        invokers: {
+            scrollToPanel: {
```

Comment:
This should be a modellised component rather than using DOM coordinates as an implicit model - also the range limiting should be done within the model space using a relay. See https://github.com/fluid-project/infusion/blob/master/src/components/textfieldControl/js/TextfieldSlider.js#L64 for examples of the latter.

Rather than a "scrollToPanel" invoker, the component should expose "panelIndex" and "panelCount" as model elements, with the final line "scrollLeft" implemented as a modelListener to the former.

Reply:
@amb26 Thanks for the comment! I'm not sure I totally understand though. Do you mind going through it step-by-step?

Reply:
@grrrero, a similar example that has implemented @amb26's suggestion is [the scrolling feature in the language panel of the first discovery tool](https://github.com/GPII/first-discovery/blob/master/src/js/panels.js#L665-L667). 

The `panelIndex` model value in @amb26's comment serves the same purpose as the `selectedLang` n the example. It keeps track of the current panel index. A click on arrow keys triggers a model change on the `panelIndex` value (to the `scrollIndex` value you calculated [here](https://github.com/grrrero/infusion/blob/efe9df3a808ecba82d3858ed2004c18ebef6bfc3/src/framework/preferences/js/ArrowScrolling.js#L63)). Then the model listener on `panelIndex` will perform the scrolling.

@amb26, correct me if this explanation doesn't seem right.

Reply:
@grrrero and @cindyli I think this may be similar to https://github.com/GPII/first-discovery/blob/master/src/js/nav.js

Reply:
@grrrero I've made some changes to finish up the work we started the other day to modelize panelIndex. I've submit a pull request against your branch ( https://github.com/grrrero/infusion/pull/1 ). You should review those changes and merge them into your branch if you are happy with them. You will need to add more tests to SeparatedPanelPrefsEditorTests for verifying the panelIndex after scrolling and to also trigger scrolling by using the change applier to change the panelIndex value.